### ISSUE TITLE
Fragment field metadata for underlying type

### ISSUE THREAD

**Issue by micimize** [dotansimha/graphql-code-generator#847]
**Is your feature request related to a problem? Please describe.**
I'm attempting to add helpers to the fragment classes generated in `graphql-to-dart` for assigning their fields to the original base types, but need a way to _either_:
* tell if the field is the same as on the base type
* get the field name from the base type if present (would be more powerful for field renaming)

Take the following example:
```graphql
fragment eventFields on Event {
  id { entityId, valid }
  icon
  title
  description
  calendarThrough: calendarEventsByEventId {
    calendarEvents: nodes {
      ...fullCalendarEvent
    }
  }
}
```
here, `id, icon, title, description` are directly sourced from `event`, and can all be copied, but `calenderThrough` is invalid.

**Describe the solution you'd like**
If each fragment field had an `isOnType` attribute, this could be avoided.

**Describe alternatives you've considered**
I can use `isLeaf` as a stop gap and flatten ids, or depend on the `Through` suffix convention I'm using, but both answers are fragile

**Comment by dotansimha** [comment]
Hi @micimize ! 
Not sure if I got it right. I'm not sure you'll be able to get through the document validation if `calenderThrough` is not a valid field on the type...

**Comment by micimize** [comment]
It's [alias syntax](https://graphql.org/learn/queries/#aliases) for queries. Thinking about it more, the right thing to do here is probably "add alias field metadata to queries/fragments"

**Comment by dotansimha** [comment]
@micimize 
In this PR: https://github.com/dotansimha/graphql-code-generator/pull/906, I also added `schemaFieldName` and `isAliased`. 
`name` will contains the alias if it exists, otherwise the original name.
`schemaFieldName` will be the original field name in the schema (ignoring aliases).
`isAliased` is a boolean indicates if the field is aliased or now.

**Comment by dotansimha** [comment]
Available in 0.14.2 🎉
